### PR TITLE
Added automaticTopologyRecovery to Fs2RabbitConfig

### PR DESCRIPTION
**Pull request by SheliakLyr**
TopologyRecovery feature in amqp-client is sometimes strange. I prefer to have it disabled. This PR adds the needed configuration option to Fs2RabbitConfig.

This change is not backwards compatible in Fs2RabbitConfig constructor. It is not possible to add a default - only one apply method can have defaults defined. I could try to add some additional methods to keep compability.

Also, I fixed a small compilation error (unrelated).

### CODE REVIEW

File: core/src/main/scala/dev/profunktor/fs2rabbit/config/Fs2RabbitConfig.scala [hunk 1]
```diff
@@ -57,6 +58,7 @@ object Fs2RabbitConfig {
       internalQueueSize: Option[Int],
       requestedHeartbeat: FiniteDuration = FiniteDuration(ConnectionFactory.DEFAULT_HEARTBEAT, TimeUnit.SECONDS),
       automaticRecovery: Boolean = true,
+      automaticTopologyRecovery: Boolean = true,
```
Comment (geirolz) [inline core/src/main/scala/dev/profunktor/fs2rabbit/config/Fs2RabbitConfig.scala:61]: This is the default value for the java library, I suppose, is it not?

**Comment by geirolz** [comment]
> This change is not backwards compatible in Fs2RabbitConfig constructor

This isn't a problem I'd say

File: core/src/main/scala/dev/profunktor/fs2rabbit/config/Fs2RabbitConfig.scala (hunk 1 above)
Comment (SheliakLyr) [reply core/src/main/scala/dev/profunktor/fs2rabbit/config/Fs2RabbitConfig.scala:61]: Yes, in "modern" versions of that library.

Comment (geirolz) [reply core/src/main/scala/dev/profunktor/fs2rabbit/config/Fs2RabbitConfig.scala:61]: Yea, I mean - the one we are using in fs2-rabbit at least

**Comment by geirolz** [comment]
P.S. Ignore the CI error, there is a PR to fix it and it isn't related to your changes

**Review by geirolz — APPROVED** [review]

**Review by matejcerny — APPROVED** [review]

**Pull request merged**